substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/access/JNIAccessFeature.java

```diff
@@ -184,28 +184,48 @@ private static ConditionalConfigurationRegistry getConditionalConfigurationRegis
         return (ConditionalConfigurationRegistry) ImageSingletons.lookup(JNIRuntimeAccess.JNIRuntimeAccessibilitySupport.class);
     }
 
-    private static JNICallTrampolineMethod createJavaCallTrampoline(BeforeAnalysisAccessImpl access, CallVariant variant, boolean nonVirtual) {
-        MetaAccessProvider wrappedMetaAccess = access.getMetaAccess().getWrapped();
-        ResolvedJavaField field = JNIAccessibleMethod.getCallWrapperField(wrappedMetaAccess, variant, nonVirtual);
+    private void createJavaCallTrampoline(BeforeAnalysisAccessImpl access, CallVariant variant, boolean nonVirtual) {
+        AnalysisMetaAccess metaAccess = access.getMetaAccess();
+        ResolvedJavaField field = JNIAccessibleMethod.getCallWrapperField(metaAccess.getWrapped(), variant, nonVirtual);
         access.getUniverse().lookup(field.getDeclaringClass()).registerAsReachable();
         access.registerAsAccessed(access.getUniverse().lookup(field));
-        ResolvedJavaMethod method = JNIJavaCallWrappers.lookupJavaCallTrampoline(wrappedMetaAccess, variant, nonVirtual);
-        JNICallTrampolineMethod trampoline = new JNICallTrampolineMethod(method, field, nonVirtual);
+        String trampolineName = JNIJavaCallTrampolines.getTrampolineName(variant, nonVirtual);
+        Method reflectionMethod = ReflectionUtil.lookupMethod(JNIJavaCallTrampolines.class, trampolineName);
+        // Look up the java method to ensure a JNICallTrampolineMethod gets created for it through
+        // com.oracle.svm.jni.hosted.JNINativeCallWrapperSubstitutionProcessor.lookup
+        metaAccess.lookupJavaMethod(reflectionMethod);
```

Comment:
`lookupJavaMethod` will already return an `AnalysisMethod` that you can call `access.registerAsCompiled` on, no need to do a second and third lookup below.

Reply:
Done